src/simplenlg/lexicon/XMLLexicon.java

```diff
@@ -230,51 +230,71 @@ public XMLLexicon(String languageCode, URI lexiconURI) {
 	}
 
 	/**
-	 * method to actually load and index the lexicon from a URI
-	 * 
-	 * vaudrypl removed call to addSpecialCases() and moved this
-	 * method to simplenlg.lexicon.english.XMLLexicon
-	 * 
-	 * @param uri
-	 */
-	private void createLexicon(URI lexiconURI) {
-		// initialise objects
-		words = new HashSet<WordElement>();
-		indexByID = new HashMap<String, WordElement>();
-		indexByBase = new HashMap<String, List<WordElement>>();
-		indexByVariant = new HashMap<String, List<WordElement>>();
-		// added by vaudrypl
-		indexByCategory = new EnumMap<LexicalCategory, List<WordElement>>(LexicalCategory.class);
+     * Load an XML Lexicon from an Input Stream
+     * @param language
+     *            the associated language ISO 639-1 two letter code
+     * @param lexiconStream
+     * @author amanning
+     */
+    public XMLLexicon(Language language, InputStream lexiconStream) {
```

Comment:
Now the lexicon can be populated with an input stream which is more Android-friendly.